boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,361 @@
+#include <assert.h>
+#include <math.h>
+#include <stdlib.h>
+#include "Io_Efuse.h"
+#include "configs/Io_EfuseConfig.h"
+
+#pragma GCC diagnostic ignored "-Wconversion"
+
+struct Efuse_Context
+{
+    SPI_HandleTypeDef *hspi;
+
+    GPIO_TypeDef *nss_port;
+    uint16_t nss_pin;
+
+    GPIO_TypeDef *fsob_port;
+    uint16_t fsob_pin;
+    GPIO_TypeDef *fsb_port;
+    uint16_t fsb_pin;
+
+    GPIO_TypeDef *channel0_port;
+    uint16_t channel0_pin;
+    GPIO_TypeDef *channel1_port;
+    uint16_t channel1_pin;
+
+    // The current state of the watchdog-in bit (bit 15). If the watchdog is
+    // enabled its state must be alternated at least once within the watchdog
+    // timeout period.
+    bool wdin_bit_to_set;
+};
+
+/**
+ * Calculate the parity of the SPI command to be sent to the Efuse and
+ * set/clear the parity bit. Using the XOR sum of bits method taken from:
+ * https://en.wikipedia.org/wiki/Parity_bit#Parity
+ * @param spi_command Original SPI command without the parity bit set/cleared
+ */
+static void Io_Efuse_CalculateParityBit(uint16_t *serial_data_input);
+
+/**
+ * Write SPI data to Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param efuse_spi_handle Handle to the Efuse's SPI peripheral
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the write was successful
+ *         EXIT_CODE_TIMEOUT if the SPI write timed-out
+ */
+static ExitCode Io_Efuse_WriteToEfuse(
+    uint16_t *         tx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin);
+
+/**
+ * Read SPI data from Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param rx_data Pointer to data being received from the Efuse
+ * @param efuse_spi_handle Handle to the Efuse's SPI peripheral
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_ReadFromEfuse(
+    uint16_t *         tx_data,
+    uint16_t *         rx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin);
+
+struct Efuse_Context *Io_Efuse_Create(
+    SPI_HandleTypeDef *const hspi,
+    GPIO_TypeDef *           chip_select_port,
```

Comment:
These should be changed to nss port and pin

Reply:
changed